apps/src/lib/kits/maker/ui/SetupInstructions.jsx

```diff
@@ -82,16 +91,112 @@ class ConnectionInstructions extends React.Component {
           <li>{applabI18n.makerSetupWebSerialConnectToComputer()}</li>
           <li>{applabI18n.makerSetupWebSerialConnectToBoardButton()}</li>
           <li>
-            <SafeMarkdown
-              markdown={applabI18n.makerSetupWebSerialWindowConnect()}
-            />
+            {applabI18n.makerSetupWebSerialWindowConnect()}
+            <ul className={styles.troubleshootList}>
+              <li>
+                <SafeMarkdown
+                  markdown={applabI18n.makerSetupNoCompatibleDevices({
+                    installInstructionsId: styles.installInstructionsId,
+                  })}
+                />
+              </li>
+            </ul>
           </li>
           <li>{applabI18n.makerSetupWebSerialSuccessfulConnection()}</li>
         </ol>
         <SafeMarkdown
           markdown={applabI18n.makerSetupWebSerialSupportArticle()}
+          openExternalLinksInNewTab={true}
         />
         {connectionState}
+        <CPExpressInstallInstructions />
+      </div>
+    );
+  }
+}
+
+class CPExpressInstallInstructions extends React.Component {
```

Comment:
You could consider using subrender functions within `CPExpressInstallInstructions` to help parse this pretty long`render` function. - maybe for step 1 and step 2?

Reply:
done!